allensdk/brain_observatory/behavior/trials_processing.py

```diff
@@ -191,6 +178,23 @@ def get_trials(data, licks_df, rewards_df, rebase):
         change_time = event_dict.get(('stimulus_changed', '')) or event_dict.get(('sham_change', '')) if stimulus_change or sham_change else float('nan')
         trial_data['change_time'].append(change_time)
 
+        # Trial type logic:
```

Comment:
Refactor of trial logic, I think this fixes the bug referred to in #792 
@nickponvert what do you think?

Reply:
Yes I think this is the right way to do this, and that it would solve #792 . This is cleaner and makes more sense to read.